vars/promoteRepos.groovy

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+
+    String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+    def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
+
+    String filename = inputManifest.build.getFilename()
+    String jobname = args.jobName
+
+    String buildnumber = args.buildNumber ?: 'none'
+    if (buildnumber == 'none') {
+        println('User did not enter build number in jenkins parameter, exit 1')
+        System.exit(1)
+    }
+
+    def distTypeMap = [
+        "yum": "rpm",
+        "apt": "deb",
+    ]
+    String repoType = args.distributionRepoType
+    String distType = distTypeMap["${repoType}"]
+    String version = inputManifest.build.version
+    String majorVersion = version.tokenize('.')[0]
+    String repoVersion = majorVersion + '.x'
+    String qualifier = inputManifest.build.qualifier ? '-' + inputManifest.build.qualifier : ''
+    String revision = version + qualifier
+    println("Product: ${filename}")
+    println("Build Number: ${buildnumber}")
+    println("Input Manifest: ${manifest}")
+    println("Revision: ${revision}")
+    println("Major Version: ${majorVersion}")
+    println("Repo Type: ${repoType}")
+    println("Dist Type: ${distType}")
+    println("Repo Version: ${repoVersion}")
+
+    String stagingPkgPathX64 = "${PUBLIC_ARTIFACT_URL}/${jobname}/${revision}/${buildnumber}/linux/x64/${distType}/dist/${filename}/${filename}-${revision}-linux-x64.${distType}"
+    String stagingPkgPathARM64 = "${PUBLIC_ARTIFACT_URL}/${jobname}/${revision}/${buildnumber}/linux/arm64/${distType}/dist/${filename}/${filename}-${revision}-linux-arm64.${distType}"
+
+    String localPath = "${WORKSPACE}/artifacts"
+    String RepoProdPath = "releases/bundle/${filename}/${repoVersion}/${repoType}"
+    String artifactPath = "${localPath}/${RepoProdPath}"
+
+    withCredentials([string(credentialsId: 'jenkins-artifact-promotion-role', variable: 'ARTIFACT_PROMOTION_ROLE_NAME'),
+        string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT'),
+        string(credentialsId: 'jenkins-artifact-production-bucket-name', variable: 'ARTIFACT_PRODUCTION_BUCKET_NAME')]) {
+            withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
+                println("Pulling Prod ${repoType}")
+                sh("aws s3 sync s3://${ARTIFACT_PRODUCTION_BUCKET_NAME}/${RepoProdPath}/ ${artifactPath}/ --no-progress")
```

Comment:
Is it safe to use `sync` or do we want to `cp`? Sync can copy unnecessary files too so copy is always a safe option

Reply:
Sync is better than cp here so we dont override existing artifacts.